flocker/control/test/test_model.py

```diff
@@ -199,6 +202,29 @@ class NodeStateTests(SynchronousTestCase):
     """
     Tests for ``NodeState``.
     """
+    def test_iclusterstatechange(self):
+        """
+        ``NodeState`` instances provide ``IClusterStateChange``.
+        """
+        self.assertTrue(
+            verifyObject(IClusterStateChange, NodeState(hostname=u"1.2.3.4"))
+        )
+
+    def test_update_cluster_state(self):
```

Comment:
Maybe have this merge old and new NodeStates (e.g. old has manifestations, new has applications and ignoreant of manifestations) to prove it's not just replacing the old NodeState with new one?